### PR TITLE
Modify on-http to use taskgraph's gRPC interface

### PR DESCRIPTION
**Pull request by brianparry**
This PR contains the following changes:
- Proxy on-http calls through on-taskgraph gRPC mechanism and remove explicit use of workflow data model
- Add optional service discovery using Consul

jenkins: depends on https://github.com/RackHD/on-taskgraph/pull/215

### CODE REVIEW

**Comment by brianparry** [comment]
test this please

**Comment by dalebremner** [comment]
jenkins: test this please

**Comment by brianparry** [comment]
test this please

**Comment by brianparry** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by BillyAbildgaard** [comment]
test this please

**Comment by dalebremner** [comment]
jenkins: test this please

**Comment by srinia6** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by BillyAbildgaard** [comment]
test this please

**Comment by BillyAbildgaard** [comment]
test this please

**Comment by BillyAbildgaard** [comment]
test this please

**Comment by brianparry** [comment]
jenkins: test this please

**Comment by BillyAbildgaard** [comment]
ESXI install was only failure.

jenkins, test this please

**Comment by johren** [comment]
Just double-checking, were the permission changes for lib/api/2.0/workflowGraphs.js and lib/api/2.0/workflowTasks.js intended?

**Comment by brianparry** [comment]
@johren Those changes were not intentional.  @BillyAbildgaard is fixing them now.

File: spec/lib/services/workflow-api-service-spec.js [hunk 1]
```diff
@@ -71,226 +64,83 @@ describe('Http.Services.Api.Workflows', function () {
...
+        this.sandbox.stub(taskGraphService, 'workflowsDeleteTasksByName');
     });
 
     afterEach('Http.Services.Api.Profiles afterEach', function() {
         this.sandbox.restore();
     });
 
     after(function() {
-        sinon.stub(workflowApiService, 'createAndRunGraph');
+        //sinon.stub(workflowApiService, 'createAndRunGraph');
```
Comment (johren) [inline spec/lib/services/workflow-api-service-spec.js]: Should this commented out line be removed?

File: spec/lib/services/workflow-api-service-spec.js [hunk 2]
```diff
@@ -71,226 +64,83 @@ describe('Http.Services.Api.Workflows', function () {
...
-            expect(eventsProtocol.publishProgressEvent).to.have.been.calledOnce;
-            expect(eventsProtocol.publishProgressEvent)
-                .to.have.been.calledWith(graph.instanceId, data);
-            expect(workflowApiService.runTaskGraph).to.have.been.calledOnce;
-            expect(workflowApiService.runTaskGraph)
-                .to.have.been.calledWith(graph.instanceId, 'test');
         });
     });
 
     it('should create and run a graph against a node', function () {
```
Comment (johren) [inline spec/lib/services/workflow-api-service-spec.js]: Several unittests removed here?  Are these no longer valid?

File: lib/services/scheduler.proto [hunk 3]
```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
```
Comment (johren) [inline lib/services/scheduler.proto]: Does this file need a copyright?

File: spec/lib/services/workflow-api-service-spec.js (hunk 2 above)
Comment (brianparry) [inline spec/lib/services/workflow-api-service-spec.js]: @johren These tests have been moved to the on-taskgraph.

**Comment by brianparry** [comment]
@johren There should be no externally visible change in behavior when this PR is merged.  This cleans up the interface between on-http and on-taskgraph and allows on-taskgraph to run as a standalone service.

**Comment by johren** [comment]
Not quite as many commits in this one, but still might be nice to squash.

**Comment by brianparry** [comment]
last commit failed jenkins due to network issues.  jenkins: test this please

**Pull request merged**